XamarinCommunityToolkit/Views/CameraView/FormsCameraView.ios.cs

```diff
@@ -0,0 +1,553 @@
+					transform = CGAffineTransform.MakeTranslation(imageSize.Height, 0);
+					transform = CGAffineTransform.Rotate(transform, (float)Math.PI / 2.0f);
+					break;
+				default:
+					throw new Exception("Invalid image orientation");
+			}
+
+			UIGraphics.BeginImageContext(bounds.Size);
+			var context = UIGraphics.GetCurrentContext();
+
+			if (orient == UIImageOrientation.Right)
+			{
+				context.ScaleCTM(-1, 1);
+				context.TranslateCTM(-imgHeight, 0);
+			}
+			else
+			{
+				context.ScaleCTM(1, -1);
+				context.TranslateCTM(0, -imgHeight);
+			}
+
+			context.ConcatCTM(transform);
+
+			context.DrawImage(new CGRect(0, 0, imgWidth, imgHeight), imgRef);
+			image = UIGraphics.GetImageFromCurrentImageContext();
+			UIGraphics.EndImageContext();
+			return image;
+		}
+
+		public override void Draw(CGRect rect)
+		{
+			previewLayer.Frame = rect;
+			base.Draw(rect);
+		}
+
+		public float Zoom
+		{
+			get => (float)(device?.VideoZoomFactor ?? 1f);
+			set
+			{
+				if (device == null)
+					return;
+				_ = device.LockForConfiguration(out _);
+				device.VideoZoomFactor = Math.Max(1, Math.Min(value, MaxZoom));
+				device.UnlockForConfiguration();
+			}
+		}
+
+		public float MaxZoom => (float)(device?.ActiveFormat.VideoMaxZoomFactor ?? 1f);
+
+		public async Task TakePhoto()
+		{
+			if (isBusy || device == null || videoOutput != null)
+				return;
+
+			IsBusy = true;
+			// iOS >= 10
+			if (photoOutput != null)
+			{
+				var photoOutputConnection = photoOutput.ConnectionFromMediaType(AVMediaType.Video);
+				if (photoOutputConnection != null)
+					photoOutputConnection.VideoOrientation = previewLayer.Connection.VideoOrientation;
+
+				var photoSettings = AVCapturePhotoSettings.Create();
+				photoSettings.FlashMode = (AVCaptureFlashMode)flashMode;
+				photoSettings.IsHighResolutionPhotoEnabled = true;
+
+				var photoCaptureDelegate = new PhotoCaptureDelegate
+				{
+					OnFinishCapture = (data, error) =>
+					{
+						FinishCapture?.Invoke(this, new Tuple<NSObject, NSError>(data, error));
+						IsBusy = false;
+					},
+					WillCapturePhotoAnimation = () => Animate(0.25, () => previewLayer.Opacity = 1)
+				};
+
+				photoOutput.CapturePhoto(photoSettings, photoCaptureDelegate);
+				return;
+			}
+			// iOS < 10
+			try
+			{
+				var connection = imageOutput.Connections[0];
+				connection.VideoOrientation = previewLayer.Connection.VideoOrientation;
+				var sampleBuffer = await imageOutput.CaptureStillImageTaskAsync(connection);
+				var imageData = AVCaptureStillImageOutput.JpegStillToNSData(sampleBuffer);
+				FinishCapture?.Invoke(this, new Tuple<NSObject, NSError>(imageData, null));
+			}
+			catch (Exception)
+			{
+				FinishCapture?.Invoke(this, new Tuple<NSObject, NSError>(null, new NSError(new NSString("faled create image"), 0)));
+			}
+			IsBusy = false;
+		}
+
+		string ConstructVideoFilename()
+		{
+			var documents = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
```

Comment:
In the future, we want to allow the dev to save the media in another folder? I can see an enhancement issue for it in the future hehe